task/update-infra-deployments/0.1/update-infra-deployments.yaml

```diff
@@ -249,23 +256,37 @@ spec:
 
             def create_reset_branch(self):
                 branch = ORIGIN_REPO.split('/')[-1]
-                target_branch = self._request("GET", f"/repos/{TARGET_GH_REPO}/git/refs/heads/{branch}").json()
-                main_branch_sha = self._request("GET", f"/repos/{TARGET_GH_REPO}/git/refs/heads/main").json()['object']['sha']
-                if "ref" in target_branch:
+
+                target_branch_sha, target_status_code = self._get_branch_ref_sha(branch)
+                main_branch_sha, main_status_code = self._get_branch_ref_sha("main")
+                response = Response(content="", status=-1)
+
+                if target_status_code == 200:
+                    # successive updates to the infra-deployments PR could result in a race condition
+                    # this should prevent the creation of PRs that update references to an older version
+                    if target_branch_sha != main_branch_sha:
+                        logger.error(f"HEAD of existing target branch {branch} is '{target_branch_sha}', main is '{main_branch_sha}'")
+                        raise RuntimeError(f"The target branch {branch} is out of sync with main")
```

Comment:
Sometimes the task is supposed to update a PR that's already open, e.g. here: https://github.com/redhat-appstudio/infra-deployments/pull/6508

One PR gets merged and triggers the creation of an infra-deployments PR.
A second PR gets merged and updates the already existing infra-deployments PR.

Will this continue to work?